utils/tools.js

```diff
@@ -146,10 +146,10 @@ var SLTools = {
             );
             results = results.concat(pageResults);
           }
-          if (page.id) {
-            page = await messenger.messages.continueList(page.id);
-          }
-        } while (page.id);
```

Comment:
Oh my, yes! 

`page.id` in the while statement is the `page.id` of the NEXT page, so we always skip the last page.

Reply:
Due to the fact that `!page.id` can only `true` if there is no `page.id` set (`null` or `undefined`), we can resolve this comment.